lib/fdotypes.c

```diff
@@ -5471,6 +5471,110 @@ bool fdo_serviceinfo_modules_list_write(fdow_t *fdow) {
 	return ret;
 }
 
+/**
+ * Return bool value representing whether any external ServiceInfo module has message to
+ * send in the NEXT iteration. This determines the TO2.DeviceServiceInfo.IsMoreServiceInfo value
+ * for any currently active module.
+ *
+ * @param fdow - Pointer to the writer.
+ * @param module_list - Pointer to the ServiceInfo module list containing all
+ * Device ServiceInfos modules.
+ * @param mtu - MTU to be used for fitting the values.
+ *
+ * @return true if there is ServiceInfo to be sent in the next iteration,
+ * false on any other conditions.
+ */
+bool fdo_serviceinfo_external_mod_is_more(fdow_t *fdow,
+	fdo_sdk_service_info_module_list_t *module_list, size_t mtu) {
+
+	if (!fdow || !module_list) {
+		return false;
+	}
+	fdo_sdk_service_info_module_list_t *traverse_list = module_list;
+	bool is_more = false;
+
+	while (traverse_list) {
+		if (traverse_list->module.active &&
+			traverse_list->module.service_info_callback(
+			FDO_SI_IS_MORE_DSI, NULL, fdow, NULL, NULL, &is_more, mtu) != FDO_SI_SUCCESS) {
+			LOG(LOG_DEBUG, "Sv_info: %s's CB Failed for type:%d\n",
+			    traverse_list->module.module_name, FDO_SI_HAS_MORE_DSI);
+			return false;
```

Comment:
Should this be handled differently that a is_more == false condition? In other words, do we need to do something more than log an error?

Reply:
Updated to store the 'is_more' in a separate parameter, and the method's return value will be treated as method's execution success/failure. The caller in msg68 will error out if there's any error.